src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -174,7 +174,7 @@ const SignOrExecuteForm = ({
           onFormSubmit={onAdvancedSubmit}
         />
 
-        {(error || (willExecute && gasLimitError)) && (
+        {(error || gasLimitError) && (
```

Comment:
While writing the test for this I noticed that we already use `willExecute` in the `useGasLimit` hook to prevent estimation if the user is not executing so no need to have this additional check here.